ThePensionsRegulator.Frontend/HtmlGeneration/ComponentGenerator.TprHeaderBar.cs

```diff
@@ -83,9 +85,96 @@ public virtual TagBuilder GenerateTprHeaderBar(TprHeaderBar tprHeaderBar)
                     contentElement.InnerHtml.Append(tprHeaderBar.Content.ToString()!);
                 }
             }
+            else if (tprHeaderBar.DisplaySearchBar)
+            {
+
+                var headerSearch = GenerateTprHeaderSearch(tprHeaderBar);
+                headerContent.InnerHtml.AppendHtml(headerSearch);
+            }
+
             tagBuilder.InnerHtml.AppendHtml(headerContent);
 
             return tagBuilder;
         }
+
+        public virtual TagBuilder GenerateTprHeaderSearch(TprHeaderBar tprHeaderBar)
+        {
+            var divTag = new TagBuilder("div");
+            if (tprHeaderBar.SearchAttributes != null) { divTag.MergeAttributes(tprHeaderBar.SearchAttributes); }
+            divTag.AddCssClass("tpr-header-search");
+
+            var form = new TagBuilder("form");
+            form.Attributes.Add("action", $"{tprHeaderBar.ActionPath}");
+            form.Attributes.Add("id", "tpr-header-search__form");
+            form.Attributes.Add("method", "get");
+
+            var searchField = new TagBuilder("div");
+            searchField.AddCssClass("tpr-header-search__field");
+
+            var autoComplete = new TagBuilder("div");
+            autoComplete.AddCssClass("tpr-header-search-autocomplete");
+            searchField.InnerHtml.AppendHtml(autoComplete);
+
+            var autoCompleteContainer = new TagBuilder("div");
+            autoCompleteContainer.AddCssClass("tpr-autocomplete-container");
+            autoCompleteContainer.Attributes.Add("autocomplete-url", tprHeaderBar.AutoCompleteUrl);
+            autoCompleteContainer.Attributes.Add("placeholder", tprHeaderBar.SearchPlaceholderText);
+            autoComplete.InnerHtml.AppendHtml(autoCompleteContainer);
+
+            var searchInput = new TagBuilder("input");
+            searchInput.Attributes.Add("aria-label", tprHeaderBar.SearchAriaLabel);
+            searchInput.Attributes.Add("placeholder", tprHeaderBar.SearchPlaceholderText);
```

Comment:
`SearchPlaceholderText` is a nullable property so this could generate `placeholder=""`. Either `SearchPlaceholderText` should be `required`, or you should avoid generating an attribute if it's blank.